packages/expo-gl/package.json

```diff
@@ -2,8 +2,8 @@
   "name": "expo-gl",
   "version": "8.0.0",
   "description": "Provides GLView that acts as OpenGL ES render target and gives GL context object implementing WebGL 2.0 specification.",
-  "main": "build/index.js",
-  "types": "build/index.d.ts",
+  "main": "build/src/index.js",
```

Comment:
Yep. It's caused by the following change in `packages/expo-gl/src/GLView.tsx`:

```diff
- const packageJSON = require('../package.json');
+ import packageJSON from '../package.json';
```

Because of this and how typescript emits, the emitted package.json appears in the toplevel directory (`build/`) and the rest of the source must be one level deeper, so it looks like tsc automatically chose `src` as the subdirectory name.

This conversion was necessary because the import order linter warned about a relative import being out of order when it was places below other imports, but when placed above other imports a different linter warned that a const assignment appeared above an import (code appearing above import). Alternatively, we could disable the linter for this file, but from reading README the philosophy seemed to be fairly against the disabling.

Reply:
This makes sense but I don't think that shipping code under `build/src` for some packages (specifically ones that import package.json) and under `build` for others is a good idea. It's going to create confusion when different packages are structured differently, and when adding or removing the import can completely change a package's build path!

Philosophically, the intent of the linter is to reduce the attention cost of writing code. If the linter ends up adding new warnings that we need to disable except in very rare cases, we're just moving attention cost around instead of reducing it. Ideally we'd be able to keep the old `require()` call in this case.

The ESLint import order plugin says, "Statements using the ES6 import syntax must appear before any require() statements." This sounds different than the behavior you were seeing, but if we could get the import order plugin to have this behavior it sounds like we'd be able to use `require()` and address this issue.

Reply:
https://github.com/benmosher/eslint-plugin-import/pull/1626